programs/aes/aescrypt2.c

```diff
@@ -78,6 +79,19 @@ int main( void )
     return( 0 );
 }
 #else
+
+#if defined(MBEDTLS_CHECK_PARAMS)
+#include "mbedtls/platform_util.h"
+void mbedtls_param_failed( const char *failure_condition,
```

Comment:
I'm not convinced by having copies of mbedtls_param_failed in so many places...

Reply:
The implementation of `mbedtls_param_failed()` *must* be provided by the application code, and I don't think belongs in the library as we can't do anything about it. 

Therefore we could have a common implementation for sample programs, but sample programs are supposed to be complete and self-contained, so we end up with one for each sample app.

It's not ideal.

Reply:
To be fully honest, I'm not sure I'm happy about that either, but that's a conscious design decision, it works well enough, so I'm not changing it at this point.